integration/src/main.rs

```diff
@@ -169,98 +191,143 @@ fn run_errors_test(
-fn run_examples_test(client: &mut postgres::Client) -> Result<bool, Box<dyn std::error::Error>> {
-    let mut successful = true;
-    let original_pwd = std::env::current_dir().unwrap();
-    for file in std::fs::read_dir("../examples")? {
+    for file in std::fs::read_dir("fixtures/codegen")? {
         let file = file?;
         let name = file.file_name().to_string_lossy().to_string();
-        let path = file.path();
-
-        print!("{} {}", "[example]".magenta(), name.magenta());
-
-        std::env::set_current_dir(path)?;
-        // Reset db
-        reset_db(client)?;
-
-        // Run codegen
-        cornucopia::run_migrations(client, "migrations")?;
-        cornucopia::generate_live(
-            client,
-            "queries",
-            Some("src/cornucopia.rs"),
-            CodegenSettings {
-                is_async: !name.contains("sync"),
-                derive_ser: false,
-            },
-        )?;
-
-        // Run example
-        let result = Command::new("cargo").arg("run").output()?;
-        if !result.status.success() {
-            successful = false;
-            println!(
-                " {}\n{}",
-                "ERR".red(),
-                String::from_utf8_lossy(&result.stderr)
-                    .as_ref()
-                    .bright_black()
-            );
+        let content = std::fs::read_to_string(file.path())?;
+        println!("{} {}", "[codegen]".magenta(), name.magenta());
+
+        let suite: CodegenTestSuite = toml::from_str(&content)?;
+
+        // If we're doing a global run of this codegen, only run the migrations once
+        let local_run = if let Some(run) = &suite.run {
+            std::env::set_current_dir(format!("../{}", run.path))?;
+            let migrations = cornucopia::read_migrations("migrations")?;
+            cornucopia::run_migrations(client, migrations)?;
+            std::env::set_current_dir(&original_pwd)?;
+            false
         } else {
-            println!(" {}", "OK".green());
+            true
+        };
+
+        for codegen_test in suite.codegen {
+            std::env::set_current_dir(format!("../{}", codegen_test.base_path))?;
+            let queries_path = codegen_test.queries.unwrap_or("queries");
+            let migrations_path = codegen_test.migrations.unwrap_or("migrations");
+            let migrations = cornucopia::read_migrations(migrations_path)?;
+            let destination = codegen_test.destination.unwrap_or("src/cornucopia.rs");
+            let is_async = !codegen_test.sync.unwrap_or(false);
+            let derive_ser = codegen_test.derive_ser.unwrap_or(false);
+
+            // If we're doing a global run of this codegen, only run the migrations once
+            if local_run {
+                cornucopia::run_migrations(client, migrations)?
+            };
+
+            // If `--apply`, then the code will be regenerated.
+            // Otherwise, it is only checked.
+            if apply {
```

Comment:
Codegen apply should have its own flag so that I can apply a new codegen without updating error messages.